sdk/core/System.ClientModel/api/System.ClientModel.net6.0.cs

```diff
@@ -116,15 +122,23 @@ public partial class PipelineMessage : System.IDisposable
     {
         protected internal PipelineMessage(System.ClientModel.Primitives.PipelineRequest request) { }
         public bool BufferResponse { get { throw null; } set { } }
-        public System.Threading.CancellationToken CancellationToken { get { throw null; } set { } }
+        public System.Threading.CancellationToken CancellationToken { get { throw null; } protected internal set { } }
+        public System.ClientModel.Primitives.PipelineMessageClassifier? MessageClassifier { get { throw null; } set { } }
         public System.TimeSpan? NetworkTimeout { get { throw null; } set { } }
         public System.ClientModel.Primitives.PipelineRequest Request { get { throw null; } }
         public System.ClientModel.Primitives.PipelineResponse? Response { get { throw null; } protected internal set { } }
+        public void Apply(System.ClientModel.Primitives.RequestOptions options) { }
         public void Dispose() { }
         protected virtual void Dispose(bool disposing) { }
         public void SetProperty(System.Type type, object value) { }
         public bool TryGetProperty(System.Type type, out object? value) { throw null; }
     }
+    public partial class PipelineMessageClassifier
```

Comment:
Pri 0: I thought we would keep this class abstract (so that it can not have fields) and just add a factory method for an internal subclass.

Reply:
Let me look - if it's feasible, happy to fix.

Reply:
Looks like it's not abstract in Core, which is why we probably did it this way.  I think I can make it abstract in ClientModel, so I'll add that to this PR.

Reply:
Actually, @KrzysztofCwalina, if we make it abstract, the default Azure.Core ResponseClassifier will have to duplicate the default implementation, or, we'd have to wrap the default instance in the Azure.Core classifier.  Do you prefer this over keeping this non-abstract and having the implementation in one place/not adding memory overhead to put the field in the Azure.Core classifier?

Reply:
Ooops, I looked at APIs of PMC and implementation of RMC.

Reply:
Discussed offline - @KrzysztofCwalina is ok with this approach.